prototypes/resources.lua

```diff
@@ -1,7 +1,7 @@
 ----------------------------------------------------------------
 -------------------------DEEP SEA OIL --------------------------
 ----------------------------------------------------------------
-if settings.startup["deep_oil"].value then
+if settings.startup["deep_oil"].value and not mods["nullius"] then
```

Comment:
This should be handled by adding Nullius to the list of mods in `settings-updates.lua` which cause the oil setting to be forced to false.